src/Uno.Extensions.Templates/content/unoapp-extensions/azure/steps/desktop.yml

```diff
@@ -0,0 +1,75 @@
+parameters:
+- name: package_id
+  type: string
+  default: 'release_Desktop'
+- name: project_to_build
+  type: string
+  default: ''  
+- name: project_folder
+  type: string
+  default: ''    
+- name: project_platform
+  type: string
+  default: 'x64'  
+- name: publish_artifacts
+  type: string
+  default: 'true'    
+- name: compress_artifacts
+  type: string
+  default: 'true' 
+
+steps:
+    #Validate required parameters before executing any other script
+    - task: PowerShell@2
+      displayName: 'Verify required parameters'
+      inputs:
+        targetType: 'inline'
+        script: |
+            $errors = ''
+            if ('' -eq '${{ parameters.project_to_build }}') { $errors = $errors + '[project_to_build] is not valid.' + "`r`n"  }
+            if ('' -eq '${{ parameters.project_platform }}') { $errors = $errors + '[project_platform] is not valid' + "`r`n"  }
+            
+            if (-not ([string]::IsNullOrEmpty($errors))){
+                $errors
+                throw "Some required parameters are missing"
+            }
+
+    - template: gitversion.yml
+
+    - task: PowerShell@2
+      displayName: 'Build and Package'
+      inputs:
+        targetType: 'inline'
+        script: |
+            $projectFolder = '${{ parameters.project_folder }}'
+            if (('' -eq $projectFolder) -and ("${{ parameters.project_to_build }}".Contains('/') )){
+              Write-Host "Adjusting proj folder"
+              $projectFolder = "${{ parameters.project_to_build }}".SubString(0, "${{ parameters.project_to_build }}".LastIndexOf('/'))
+            }
+            Write-Host "Current proj folder is: $projectFolder"
+
+            Write-Host  ""
```

Comment:
This all seems unnecessary... the template should show what a valid input looks like.. and that shouldn't be something that changes... we should need to clean up a bad input.

Reply:
Specifying project folder is unnecessary in most scenarios since it can be inferred from the csproj. We can clean this up completely by removing the parameter. Only reason I left it was just in case they want to push to a different output folder.